core/patina_internal_cpu/src/interrupts/aarch64/gic_manager.rs

```diff
@@ -174,25 +224,33 @@ impl AArch64InterruptInitializer<'_> {
         let int_id = self.source_to_intid(interrupt_source)?;
 
         let level = if int_id.is_private() {
-            let mut sgi = self.gic_v3.sgi_ptr(0);
-            field!(sgi, icfgr).get(index).unwrap().read() & bit != 0
+            // arm-gic does not presently provide a way to directly read the interrupt state for a given interrupt.
+            // so save the current redistributor context and read the ICFGR from there.
+            let redistributor = self.gic_v3.redistributor(self.cpu_r_idx).expect("Invalid redistributor");
```

Comment:
Same logic in a few places, worth a helper func and/or upstreaming something to arm-gic?